File: Python/Easy/1323_Maximum_69_Number/solution.py
class Solution:
    def maximum69Number (self, num: int) -> int:
        output = ""
        num = str(num)
        for i in range(len(num)):
            if num[i] == "6":
                output += "9"
                if i < len(num) - 1:
                    output += num[i+1:]
                break
            else:
                output += num[i]
        return int(output)

File: Golang/Easy/1323_Maximum_69_Number/solution.go
package maximum69number

func maximum69Number(num int) int {
	numArray := []int{}

	for num%10 != 0 {
		numArray = append(numArray, num%10)
		num = num / 10
	}

	output := 0
	haveChange := false
	for i := len(numArray) - 1; i >= 0; i-- {
		if !haveChange && numArray[i] == 6 {
			output = output*10 + 9
			haveChange = true
		} else {
			output = output*10 + numArray[i]
		}
	}

	return output
}
